File: frontend/src/types/CartItem.ts
import { Product } from "./Product.ts";

export class CartItem {
    id: number;
    productCount: number;
    product: Product;
    totalAmount: number;

    constructor(id: number, product: Product,productCount: number,  totalAmount: number) {
        this.id = id;
        this.product = product;
        this.productCount = productCount;
        this.totalAmount = totalAmount;
    }
}

File: backend/src/main/java/com/wonlinestore/backend/entity/CartItem.java
package com.wonlinestore.backend.entity;

import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.math.BigDecimal;

@AllArgsConstructor
@NoArgsConstructor
@Data
@Entity
public class CartItem {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private int id;
    @ManyToOne
    private Product product;
    @Column(nullable = false)
    private int productCount;
    private BigDecimal totalAmount ;

}
